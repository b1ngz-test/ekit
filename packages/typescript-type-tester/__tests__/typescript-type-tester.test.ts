import path from 'path';
import parser from '../src';

describe('typescript-type-tester', () => {
  it('parser should detect error', () => {
    const sample = 'error.ts';
    const { [sample]: error } = parser([sample], path.join(__dirname, 'samples'));
    expect(error.length).toBeGreaterThan(0);
    expect({ [sample]: error }).toMatchSnapshot(sample);
  });
});
