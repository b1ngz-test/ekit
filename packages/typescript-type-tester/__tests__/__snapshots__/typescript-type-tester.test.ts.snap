// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`typescript-type-tester parser should detect error: error.ts 1`] = `
Object {
  "error.ts": Array [
    Object {
      "category": 1,
      "code": 2322,
      "fileName": "__tests__/samples/error.ts",
      "messageText": "Type '\\"1\\"' is not assignable to type 'number'.",
    },
  ],
}
`;
